constraints/miden-vm/hash.air

```diff
@@ -0,0 +1,76 @@
+mod HashAir
+
+### Constants and periodic columns ################################################################
+
+periodic_columns:
+    k0: [0, 0, 0, 0, 0, 0, 0, 1]
+    k1: [0, 0, 0, 0, 0, 0, 1, 0]
+    k2: [1, 0, 0, 0, 0, 0, 0, 0]
+
+### Helper functions/evaluators ###################################################################
+
+# Enforces that column must be binary.
+ev is_binary(main: [a]):
+    enf a^2 - a = 0
+
+### Bitwise Chiplet Air Constraints ###############################################################
+
+# Enforces the constraints on the hash chiplet, given the columns of the hash execution trace.
+ev hash_chiplet(main: [s[3], r, h[12], i]):
+
+    ## Instruction flags ##
+    let f_rpr = 1 - k0
+    let f_bp = k2 * s[0] * (1 - s[1]) * (1 - s[2])
```

Comment:
1. Should we rewrite this as:
`let f_bp = k2 & s[0] & !s[1] & !s[2]`

2. Should we declare an intermediate variable for `k2 & s[0]`?

Reply:
I'm not sure why should we create intermediate variables for the `k0 * (1 - s[0])` and similar expressions. I think in this case readability will be decreased since every row is described with set of selector values (e.g. `# this flag is set to 1 when flags are (i, j, k) and ... `) and in this case it will be more difficult to keep track of the value of the first flag

Reply:
I think you're imagining this in a different way from what we intended. If you were to follow this suggestion, it would be given a meaningful name, so we would have something like the following:

```
let last_row_and_s0 = k0 * s[0]
let first_row_and_s0 = k2 * s[0]
```

But looking at this, I wonder if the most readable thing to do would actually be to rename the periodic columns like:

```
periodic_columns:
    cycle_row_7: [0, 0, 0, 0, 0, 0, 0, 1]
    cycle_row_6: [0, 0, 0, 0, 0, 0, 1, 0]
    cycle_row_0: [1, 0, 0, 0, 0, 0, 0, 0]
```